pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -129,10 +130,23 @@ func (s *Server) initDNSCerts(hostname, customHost, namespace string) error {
 
 	var certChain, keyPEM, caBundle []byte
 	var err error
-	if features.PilotCertProvider == constants.CertProviderKubernetes {
+	if s.RA != nil {
```

Comment:
Given we already have a configuration "PilotCertProvider" to designate who signs Istiod DNS certs, don't think its a good idea to derive this config from anywhere else, including s.RA. Perhaps we could say that if the signerName is configured, we will use that signer to sign DNS certificates if PilotCertProvider=Kubernetes

Reply:
uf `PilotCertProvider=Kubernetes && singerName is configured ` we will not create CA now.